starknet_py/utils/data_transformer/data_transformer.py

```diff
@@ -148,16 +154,76 @@ def from_python(self, cairo_type, name, value):
 
         return results
 
+    def _from_python_named(self, cairo_type, name, value):
+        if not isinstance(value, dict) and not self._isnamedtuple(value):
+            raise ValueError(
+                f"Input {name} is a named tuple and must be dict or NamedTuple"
+            )
+
+        results = []
+        values = value
+
+        if self._isnamedtuple(values):
+            values = values._asdict()
+
+        for member in cairo_type.members:
+            result = self.resolve_type(member).from_python(
+                member, f"{name}[{member.name}]", values[member.name]
+            )
+            results.extend(result)
+
+        return results
+
+    @staticmethod
+    def _isnamedtuple(value):
+        return isinstance(value, tuple) and hasattr(value, "_fields")
+
     def to_python(self, cairo_type, name, values):
+        if not cairo_type.is_named:
+            return self._to_python_unnamed(cairo_type, name, values)
+        return self._to_python_named(cairo_type, name, values)
+
+    def _to_python_unnamed(self, cairo_type, name, values):
         result = []
         for index, member_type in enumerate(cairo_type.members):
-            transformed, values = self.resolve_type(member_type).to_python(
+            (name, transformed), values = self.resolve_type(member_type).to_python(
                 member_type, f"{name}[{index}]", values
             )
             result.append(transformed)
 
         return (*result,), values
 
+    def _to_python_named(self, cairo_type, name, values):
+        result = {}
+        for index, member_type in enumerate(cairo_type.members):
+            (name, transformed), values = self.resolve_type(member_type).to_python(
+                member_type, f"{name}[{index}]", values
+            )
+            result[name] = transformed
+
+        res = NamedTuple(
+            "Result", [(key, type(value)) for key, value in result.items()]
+        )
+        # pylint: disable=not-callable
+        return res(**result), values
+
+
+class TupleItemTransformer(TypeTransformer[TypeTuple.Item, tuple]):
+    def from_python(self, cairo_type: UsedCairoType, name: str, value: any):
+        value = self.resolve_type(cairo_type.typ).from_python(
+            cairo_type.typ,
+            cairo_type.name if "cairo_type.name" is not None else "",
+            value,
+        )
+        return value
+
+    def to_python(self, cairo_type: UsedCairoType, name: str, values: CairoData):
+        name = cairo_type.name if cairo_type.name is not None else ""
```

Comment:
Same as https://github.com/software-mansion/starknet.py/pull/109/files#r826114229

```suggestion
        name = cairo_type.name or ""
```